arch/x86/kernel/slaunch.c

```diff
@@ -553,23 +624,39 @@ static long slaunch_expose_securityfs(void)
 	}
 
 	if (sl_evtlog.addr > 0) {
-		entry--;
-		fs_entries[entry] = securityfs_create_file(sl_evtlog.name,
-					   S_IRUSR | S_IRGRP,
-					   fs_entries[SL_DIR_ENTRY], NULL,
-					   &sl_evtlog_ops);
-		if (IS_ERR(fs_entries[entry])) {
-			pr_err("Error creating securityfs %s file\n",
-				sl_evtlog.name);
-			ret = PTR_ERR(fs_entries[entry]);
+		
+		ret = sl_create_file(--entry, sl_evtlog.name, &sl_evtlog_ops);
+		if (ret)
+			goto err_dir;
+		ret = sl_create_file(--entry, "txt_sts",&sts_ops);
```

Comment:
These entries should be under a separate test for `(sl_flags & SL_FLAG_ARCH_TXT)`. It would also be more logical instead of having to prefix all the nodes with `txt_`, to create a txt directory node and place these entries under that node.

Reply:
Just to be clear, the addition of these entries should be independent/outside of the check if the log addr has been set and toggled with the test in the initial comment.